### ISSUE TITLE
Readme and tests are not up to date 

### ISSUE THREAD

**Issue by RomanBelkov** [lkl/linux#169]
Hello!

I've been playing around with LKL and here's what gave me troubles.

First of all, main page readme says you have to build using `make tools/lkl`, but `prepare3` from Makefile:988 in root does not let us build, so `cd tools` and then `make lkl` did the trick for me.

However, more confusing fact that `make test` fails when tries to work with btrfs since
```
'/tmp/tmp.5cBYz2U9vD' is too small to make a usable filesystem
Minimum size for each btrfs device is 41943040.
Makefile:43: recipe for target 'test' failed
make[1]: *** [test] Error 1
Makefile:158: recipe for target 'test' failed
make: *** [test] Error 2
```

Adjusting the count value in boot.sh:46 to a higher value (I changed 20480 to 204800) makes the test work fine.

I know that these fixes are not something great, but they can get you on track faster without losing some time. Hope this helps.

**Comment by thehajime** [comment]
@RomanBelkov thanks for the report.

> First of all, main page readme says you have to build using make tools/lkl, but prepare3 from Makefile:988 in root does not let us build, so cd tools and then make lkl did the trick for me.

could you give us a bit more information ?
- the full output of the error and paste here is helpful
- what is your distribution/OS that gives this error ?
- version of make (`make --version`)

maybe @tommie-lie could give us an answer ?

I'm also aware of the 2nd issue about btrfs on my Fedora 23.  I will fix it soon or patches/pull request is always welcome !

**Comment by RomanBelkov** [comment]
Hi @thehajime,

I used Ubuntu 16.04 with VirtualBox but got the same error using Fedora 23 running natively on another PC.

Here is the log:
```
rb@rb-VirtualBox:~/Downloads/linux-master$ make --version
GNU Make 4.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
rb@rb-VirtualBox:~/Downloads/linux-master$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04 LTS
Release:	16.04
Codename:	xenial
rb@rb-VirtualBox:~/Downloads/linux-master$ make tools/lkl
  DESCEND  lkl
rb@rb-VirtualBox:~/Downloads/linux-master$ make clean
  CLEAN   .
  CLEAN   lib/raid6
  CLEAN   lib
  CLEAN   .tmp_versions
rb@rb-VirtualBox:~/Downloads/linux-master$ make tools/lkl
  DESCEND  lkl
  HOSTCC  scripts/basic/fixdep
  HOSTCC  scripts/kconfig/conf.o
  HOSTCC  scripts/kconfig/zconf.tab.o
  HOSTLD  scripts/kconfig/conf
scripts/kconfig/conf  --defconfig Kconfig
kernel/time/Kconfig:155:warning: range is invalid
#
# configuration written to .config
#
scripts/kconfig/conf  --silentoldconfig Kconfig
kernel/time/Kconfig:155:warning: range is invalid
  CHK     include/config/kernel.release
  Using /home/rb/Downloads/linux-master as source for kernel
  /home/rb/Downloads/linux-master is not clean, please run 'make mrproper'
  in the '/home/rb/Downloads/linux-master' directory.
/home/rb/Downloads/linux-master/Makefile:988: recipe for target 'prepare3' failed
make[4]: *** [prepare3] Error 1
Makefile:146: recipe for target 'sub-make' failed
make[3]: *** [sub-make] Error 2
Makefile:124: recipe for target 'lib/lkl.o' failed
make[2]: *** [lib/lkl.o] Error 2
Makefile:60: recipe for target 'lkl' failed
make[1]: *** [lkl] Error 2
Makefile:1550: recipe for target 'tools/lkl' failed
make: *** [tools/lkl] Error 2
rb@rb-VirtualBox:~/Downloads/linux-master$ 
```

**Comment by thehajime** [comment]
thanks @RomanBelkov.

I wasn't well aware of this part of README -  I'm always building with `make -C tools/lkl` instead of `make tools/lkl`. 

@tommie-lie, what do you think ? README changes were introduced by 69a8b30faf4004ef13bd852627cffb2cd47a65b0 FYI.

**Comment by Rondom** [comment]
For me building with `make -C tools/lkl` overwrites my '.config', while 'make tools/lkl´ did/does not (if I remember correctly).

**Comment by tommie-lie** [comment]
> @tommie-lie, what do you think ? README changes were introduced by 69a8b30 FYI.

It was working then ;-) (try by checking out this commit)

The behaviour was introduced when merging upstream commits, more specifically when merging e17cf3a80d4ba0c4e40bf1a89deb1354c2e10e14. Before this patch, $O was empty if it was not set by the user. Now, $O equals $srctree which is automatically passed to tools/Makefile, then to tools/lkl/Makefile and then back to the Kernel's main Makefile. The latter does not cope well when $O == $srctree. Try e.g. calling `make mrproper && make defconfig && make O=$(pwd)`. This command has nothing to do with lkl but still fails with the same error message.

One workaround would be to override $O in tools/lkl/Makefile and pass an empty string to the Kernel's Makefile. This would work in the standard case but not if the user explicitly requested another output directory. I'm not sure about the proper place to fix this. Maybe the kernel should support $O == $srctree in which case we have to discuss this with upstream. Or lkl could completely ignore the O= setting and always build the kernel in-tree. Or we do something more sophisticated and recognize $O == $srctree. In my opinion, this is an upstream bug: either the kernel has to support building in-tree by having $O == $srctree, or the tools Makefile should never set this value for O. It is only a coincidence that e17cf3a80d4ba0c4e40bf1a89deb1354c2e10e14 did not introduce other bugs because none of the other tools directly call the kernel's Makefile.
Any comments? I can provide a pull request if the right way to do it is determined, but I doubt I have the time to discuss this with upstream/at lkml.

**Comment by Rondom** [comment]
The btrfs-issue was fixed in 8883b5f.

**Comment by thehajime** [comment]
@Rondom right.

Remaining issue is about `make tools/lkl` command, as reported @tommie-lie, but I haven't investigated yet thoughtfully .
